src/file.js

```diff
@@ -88,13 +88,19 @@ exports.write = function (fileEntry, data, options = {}) {
 exports.read = function (fileEntry, options = {}) {
   return utils.promiseCall(fileEntry, 'file')
     .then(file => {
-      return new Promise((resolve, reject) => {
-        const reader = new FileReader();
-        reader.onload = () => resolve(reader.result);
-        reader.onerror = () => reject(reader.error);
-        // see: https://developer.mozilla.org/ru/docs/Web/API/FileReader
-        readAs(options.type, reader, file);
-      });
+      if (options.type === 'Blob') {
+        return new Response(file).blob();
+      } else if (options.type === 'MutableFile') {
```

Comment:
Isn't `type='MutableFile'` redundant? It looks like it can be replaced with `fs.getEntry()`.
so we avoid additional type and keep it more simple.

Reply:
Utility function `freezeMutableFile` extracted.

`fs.getEntry` returns a `FileSystemEntry` which does not implement `File` or `Blob` and cannot be used directly. You need to `utils.promiseCall(fileEntry, 'file')` to retrieve the `File`, so `MutableFile` is not redundant.

My reason to have both `MutableFile` and `Blob` is to let user to decide the trade-off. `MutableFile` is faster, but may have a data race. `Blob` is slower but safer.

Reply:
> fs.getEntry returns a FileSystemEntry which does not implement File or Blob and cannot be used directly. You need utils.promiseCall(fileEntry, 'file') to retrieve the File, so MutableFile is not redundant.

Sure. Didn't notice that.

> My reason to have both MutableFile and Blob is to let user to decide the trade-off. MutableFile is faster, but may have a data race. Blob is slower but safer.

Ok, I think we should add this to docs, as currently I feel confused when looking on both `Blob` and `MutableFile`.
Also I suggest to name `File` instead of `MutableFile`. We can link to [MDN File documentation](https://developer.mozilla.org/en-US/docs/Web/API/File) and be consistent with browser api naming.

Reply:
Trade-off added to jsdoc.

`MutableFile` actually does *not* comply with the `File` spec:
* `File` inherits `Blob`, which means a `File` should be immutable, or at least throws error when its content has changed (this is what FF does with its `IDBMutableFile`).
* In our case, the content of a previously created `MutableFile` can be changed silently without error.

A `File` submitted with `<input>` also suffers the same problem, i.e. its content may change if the user edits the original file on the disk at the same time. Such situation is rare though. In this library however data race may occur frequently. Therefore I think it would be better to explicitly distinguish between these two types, by intentionally using a different name.

Reply:
Ok.